src/app/product/product.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include <string>
+
+class Product
+{
+  public:
+    Product() = delete;
+    Product(const std::string &name, double price, int amount);
+
+    void increaseAmount(int amount);
+    void decreaseAmount(int amount);
+    std::string_view getName() const;
+    int getAmount() const;
+
+  private:
+    std::string name_;
+    double price_;
```

Comment:
Consider making setter for this variable with validation so it won't go below 0.

Reply:
Done